examples/events/main.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/vmware/govmomi/event"
+	"github.com/vmware/govmomi/examples"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// example use: go run main.go -url $GOVC_URL -b 8h -f VmEvent
+func main() {
+	begin := flag.Duration("b", time.Hour, "Begin time") // default BeginTime is 1h ago
+	follow := flag.Bool("f", false, "Follow event stream")
+
+	examples.Run(func(ctx context.Context, c *vim25.Client) error {
+		m := event.NewManager(c)
+
+		ref := c.ServiceContent.RootFolder
+
+		filter := types.EventFilterSpec{
+			EventTypeId: flag.Args(), // e.g. VmEvent
+			Entity: &types.EventFilterSpecByEntity{
+				Entity:    ref,
+				Recursion: types.EventFilterSpecRecursionOptionAll,
+			},
+			Time: &types.EventFilterSpecByTime{
+				BeginTime: types.NewTime(time.Now().Add(*begin * -1)),
```

Comment:
Should this be `.UTC()`? (assuming vCenter uses UTC)

Reply:
Yes, you're right. But even more accurate is to use vCenter's clock.. updated the example to use the [CurrentTime](https://vdc-download.vmware.com/vmwb-repository/dcr-public/b50dcbbf-051d-4204-a3e7-e1b618c1e384/538cf2ec-b34f-4bae-a332-3820ef9e7773/vim.ServiceInstance.html#currentTime) method